lib/chef/shell.rb

```diff
@@ -295,18 +296,19 @@ def parse_opts
     private
 
     def config_file_for_shell_mode(environment)
+      dot_chef_dir = Chef::Util::PathHelper.home('.chef')
       if config[:config_file]
         config[:config_file]
-      elsif environment && ENV['HOME']
+      elsif environment && dot_chef_dir
```

Comment:
It seems like this could be a breaking change.  If the home directory didn't exist before, it would `exit 1` on line 307.  Now it will skip to the next `elsif`